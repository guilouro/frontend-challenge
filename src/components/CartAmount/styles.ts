import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Item = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.normal};
    color: ${theme.colors.text};
    margin: ${theme.spacings.xsmall} 0;

    &:nth-child(3) {
      border-top: 1px solid ${theme.colors.line};
      padding-top: ${theme.spacings.xsmall};
    }
  `}
`

export const Shipping = styled.strong`
  ${({ theme }) => css`
    color: ${theme.colors.success};
    font-size: ${theme.font.sizes.xsmall};
  `}
`
