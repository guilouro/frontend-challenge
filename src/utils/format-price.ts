export default function formatPrice(price: number | bigint): string {
  return new Intl.NumberFormat('de-DE', {
    style: 'currency',
    currency: 'EUR'
  }).format(price)
}
