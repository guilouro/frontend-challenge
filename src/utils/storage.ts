import { CartObject } from 'hooks/use-cart'

const APP_KEY = 'BARKYN_CART'

export function getStorageItem() {
  if (typeof window === 'undefined') return

  const data = window.localStorage.getItem(APP_KEY)
  return JSON.parse(data || '[]')
}

export function setStorageItem(value: CartObject) {
  if (typeof window === 'undefined') return

  const data = JSON.stringify(value)
  return window.localStorage.setItem(APP_KEY, data)
}
