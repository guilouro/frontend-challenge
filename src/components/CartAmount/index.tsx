import formatPrice from 'utils/format-price'

import * as S from './styles'

export type CartAmountProps = {
  subtotal: number
  shipping?: number
  omitShipping?: boolean
}

const CartAmount = ({
  subtotal,
  shipping = 0,
  omitShipping = false
}: CartAmountProps) => (
  <S.Wrapper>
    <S.Item>
      <span>Subtotal</span>
      <strong>{formatPrice(subtotal)}</strong>
    </S.Item>
    {!omitShipping ? (
      <>
        <S.Item>
          <span>Shipping</span>
          <S.Shipping>{shipping ? formatPrice(shipping) : 'Free'}</S.Shipping>
        </S.Item>
        <S.Item>
          <span>Total</span>
          <strong>{formatPrice(subtotal + shipping)}</strong>
        </S.Item>
      </>
    ) : null}
  </S.Wrapper>
)

export default CartAmount
