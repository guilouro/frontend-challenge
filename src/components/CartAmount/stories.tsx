import { Story, Meta } from '@storybook/react'
import CartAmount, { CartAmountProps } from '.'

export default {
  title: 'CartAmount',
  component: CartAmount,
  args: {
    subtotal: 2.9
  }
} as Meta

export const Default: Story<CartAmountProps> = (args) => (
  <CartAmount {...args} />
)
