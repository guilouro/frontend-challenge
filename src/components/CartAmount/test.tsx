import 'intl.mock'
import { render, screen } from 'utils/test-utils'

import CartAmount from '.'

describe('<CartAmount />', () => {
  it('should show free when shipping is empty', () => {
    const { container } = render(<CartAmount subtotal={2.9} />)

    expect(screen.getAllByText('2.9')).toHaveLength(2)
    expect(screen.getByText(/free/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should show shipping value', () => {
    render(<CartAmount subtotal={2.9} shipping={1.1} />)

    expect(screen.getByText('2.9')).toBeInTheDocument()
    expect(screen.getByText('1.1')).toBeInTheDocument()
    expect(screen.getByText('4')).toBeInTheDocument()
  })
})
