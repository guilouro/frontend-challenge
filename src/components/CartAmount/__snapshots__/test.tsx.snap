// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartAmount /> should show free when shipping is empty 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-size: 1.4rem;
  font-weight: 400;
  color: #383743;
  margin: 1.6rem 0;
}

.c0:nth-child(3) {
  border-top: 1px solid #e7e7e9;
  padding-top: 1.6rem;
}

.c1 {
  color: #00b856;
  font-size: 1.2rem;
}

<div
  class=""
>
  <div
    class="c0"
  >
    <span>
      Subtotal
    </span>
    <strong>
      2.9
    </strong>
  </div>
  <div
    class="c0"
  >
    <span>
      Shipping
    </span>
    <strong
      class="c1"
    >
      Free
    </strong>
  </div>
  <div
    class="c0"
  >
    <span>
      Total
    </span>
    <strong>
      2.9
    </strong>
  </div>
</div>
`
